torchmetrics/utilities/checks.py

```diff
@@ -604,3 +605,117 @@ def _check_retrieval_target_and_prediction_types(
     preds = preds.float()
 
     return preds.flatten(), target.flatten()
+
+
+def _allclose_recursive(res1: Any, res2: Any, atol: float = 1e-8):
+    """Utility function for recursively asserting that two results are within a certain tolerance."""
+    # single output compare
+    if isinstance(res1, Tensor):
+        return torch.allclose(res1, res2, atol=atol)
+    elif isinstance(res1, Sequence):
```

Comment:
```suggestion
    elif isinstance(res1, str):
        return res1 == res2
    elif isinstance(res1, Sequence):
```

special-casing strings here since they are also valid Sequences so you would run in infinite recursion issues :)